### PR TITLE
Refactor project to use smaller models

### LINKED ISSUE #57: Split data training model
# Summary
Divide the model into different compartments for increased specialization. Potentially use a top model to decide which command is sent.

# Examples
N/A

# Context
Currently, there is only one model that has to be able to determine the intent in the given text for all possible modules. The current approach limits the modifiability and extensibility of the application. If a new module is added, the big NLP model has to be trained from scratch and will become less specialized and also slower. Also all modules has to share the same labels, which makes it difficult for a module to introduce new ones.

We recommend the following approach instead. 

If we are able to determine which module is best suited to handle the given input text, we can feed that text to that module. Which in turn has a specialized NLP model just for its use-case. Its model will be able to determine the intent for its use-case and can label the text as it sees fit. This also makes it possible to develop several models simultaneously.

### PR DESCRIPTION
**Pull request by markhakansson**
# Proposed changes
* Each module now uses its own model for NLP
* CLI now runs a model that checks the text and chooses the best suited module to handle that text
* Many minor changes that I forgot

# Related issue
Fixes #57

### CODE REVIEW

**Comment by markhakansson** [comment]
Note that this is not the most beautiful refactoring. In another PR the loading of models should be done through the settings instead of hard coded.

**Review by Blinningjr — APPROVED** [review]

File: lib/automate/modules/reminder.py [hunk 1]
```diff
@@ -38,19 +40,23 @@ def notify(self, os, body):
             # TODO: Add windows call here
             pass
 
-    def run(self, _to, when, body, _sender):
+    def run(self, text, sender):
+        to, when, body = self.nlp(text)
+        return self.execute_task(to, when, body, sender)
+
+    def execute_task(self, _to, when, body, _sender):
```
Comment (97gushan) [inline lib/automate/modules/reminder.py:47]: perhaps remove sender as a parameter from execute_task if it isn't used??

**Review by 97gushan — COMMENTED** [review]
One minor possible change but looks good otherwise

File: lib/automate/modules/reminder.py (hunk 1 above)
Comment (markhakansson) [reply lib/automate/modules/reminder.py:47]: I would say it is up for the one responsible for this module. I did not feel comfortable refactoring the code itself as I've barely been working with it.

File: scripts/install_nlp_models.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (viktorfrom) [inline scripts/install_nlp_models.sh:1]: This script is no longer needed, it has been resolved with the latest 'source substorm.sh' script.

**Review by 97gushan — APPROVED** [review]

**Pull request merged**